exporter/splunkhecexporter/client.go

```diff
@@ -66,45 +76,47 @@ func (c *client) pushMetricsData(
 	c.wg.Add(1)
 	defer c.wg.Done()
 
-	splunkDataPoints, _ := metricDataToSplunk(c.logger, md, c.config)
-	if len(splunkDataPoints) == 0 {
-		return nil
-	}
+	gzipWriter := c.zippers.Get().(*gzip.Writer)
+	defer c.zippers.Put(gzipWriter)
 
-	body, compressed, err := encodeBodyEvents(&c.zippers, splunkDataPoints, c.config.DisableCompression)
-	if err != nil {
-		return consumererror.NewPermanent(err)
-	}
+	gzipBuffer := bytes.NewBuffer(make([]byte, 0, c.config.MaxContentLengthLogs))
+	gzipWriter.Reset(gzipBuffer)
 
-	req, err := http.NewRequestWithContext(ctx, "POST", c.url.String(), body)
-	if err != nil {
-		return consumererror.NewPermanent(err)
-	}
+	// Callback when each batch is to be sent.
+	send := func(ctx context.Context, buf *bytes.Buffer, headers map[string]string) (err error) {
```

Comment:
Passing headers with the callback shouldn't be needed for metrics

Reply:
it is needed to allow configuration of the passthrough hec token.

Reply:
Can you point me to the code where headers passed to the callback? I cannot find where `send` is called with anything than `nil` headers

Reply:
Oh boy, ok. Removed.